include/h2o/pipe_utils.h

```diff
@@ -0,0 +1,39 @@
+#ifndef __H2O__PIPE_UTILS__H__
+#define __H2O__PIPE_UTILS__H__
+
+#include <stddef.h>
+#include <stdbool.h>
+
+enum h2o_send_state;
+struct st_h2o_req_t;
+struct st_h2o_context_t;
+
+typedef struct st_h2o_pipe_reader_t {
+    int fds[2]; /* fd[0] set to -1 unless used */
+    int inflight;
+
+    size_t body_bytes_read, body_bytes_sent;
+} h2o_pipe_reader_t;
+
+int h2o_empty_pipe(int fd);
```

Comment:
I am not sure if we can have a function that clears any pipes. The pipe reader can clear it because it knows how data is written to the pipe.

Considering that, IMO it would make sense to:
* turn this function into a static function inside .c, and
* rename the added files to pipe_reader.[ch], as they would then deal only with pipe readers.

Reply:
Thank you, fixed in e34e5903b538548260b59ca2db646e6d6720734c